monitoring rework (#265)
Xceptance/xlt-documentation: content/en/xtc/monitoring/420-monitoring-configuration.md

{{< /image >}}

## Notification Lists

XTC uses a flexible approach to configuring when an alert is triggered and to whom notifications are sent. Instead of having a single list of recipients where all recipients are notified in every case, you can define your own lists of who gets notified and how (**notification lists**) and under which circumstances these alerts should be sent ([**quality sensors**]({{< relref "#quality-sensors" >}})). 

Notification lists are managed globally for a project, on the project configuration's _Notification Lists_ tab. 

In a simple project, the setup of notification lists might look like this:

* **On-Call Team**: Evaluates scenario failures and takes necessary actions (perhaps alerting other people).
* **Monitoring Dev Team**: Maintains the monitoring scenarios.
* **App Dev Team**: Addresses performance issues in the monitored application.

The default **Reply-To Address** for all notification e-mails can be defined in the _General_ settings on the _Notification Lists_ tab. If none is set this will be a no-reply service address, so we recommend using any sensible contact in your project for this.

### Defining a Notification List

> To add a new notification list, go to the _Quality Sensors_ tab in _Configuration_ and click _New_ at the top right. You will be asked to set a name for your list and, if needed, add a description.

Reviewer: ```suggestion
To add a new notification list, go to the _Notification Lists_ tab in _Configuration_ and click _New_ at the top right. You will be asked to set a name for your list and, if needed, add a description.
```

The newly created notification list will then show up in the overview table. 

{{< image src="xtc/monitoring_notificationLists_overview.png" >}}
Notification lists overview table.
{{< /image >}}

By clicking its context menu you may **edit** the name and description defined before, **duplicate** it in order to create another similar list or **delete** it (you will be asked for confirmation). You can also [**pause or disable**]({{< relref "#pausing-or-disabling-notifications" >}}) notifications for a single list here. 

To view and edit the subscribers and check assignments of a notification list, either click the list's name or select **view** from the context menu. 

{{< image src="xtc/monitoring_notificationLists_details.png" >}}
Notification list detail view.
{{< /image >}}

> To add new subscribers to a notification list, got to the _Subscribers_ tab in the list detail view and click _Add_. You may add **XTC user subscribers** (existing XTC account with predefined contact data) or **custom subscribers** (custom mail address or phone number). For each subscriber you may choose whether to send notifications via e-mail or text message (or both). You can also add a [**slack channel**]({{< relref "../integrations/510-slack/" >}}) to the subscribers list. The subscriber data can be edited anytime, subscribers can also be deactivated or removed entirely.

Reviewer: ```suggestion
To add new subscribers to a notification list, go to the _Subscribers_ tab in the list detail view and click _Add_. You may add **XTC user subscribers** (existing XTC account with predefined contact data) or **custom subscribers** (custom mail address or phone number). For each subscriber you may choose whether to send notifications via e-mail or text message (or both). You can also add a [**slack channel**]({{< relref "../integrations/510-slack/" >}}) to the subscribers list. The subscriber data can be edited anytime, subscribers can also be deactivated or removed entirely.
```

### Notification List Assignments

When you are finished configuring a notification list, you can [use it in quality sensors]({{< relref "#defining-a-quality-sensor" >}}). You can check which quality sensors the notification list is assigned to in the notification list's details page on the _Assignments_ tab. You can use this to check which alerts you might affect by changing any notification list.

### Pausing or Disabling Notifications

> On the project configuration's _Notification Lists_ tab, you can globally **disable** and enable notifications for the whole project. 

Reviewer: ```suggestion
On the project configuration's _Notification Lists_ tab, you can **globally disable or enable** notifications for the whole project. 
```

To **pause** notifications (and, if you configure this, pause scenario execution) for a predefined period in time you may configure [**Quiet Periods**]({{< relref "#quiet-periods" >}}).

To enable or disable notifications or define notification pauses for any **single notification list** you can use this list's context menu in the notification lists overview. The list's notification state will be indicated by a symbol in the overview table.

{{< image src="xtc/monitoring_notificationLists_pausedDisabled.png" >}}
Notification lists that (temporarily) won't receive notifications.
{{< /image >}}

## Quality Sensors 

A **quality sensor** defines the circumstances under which an alert is to be sent and [which user groups]({{< relref "#notification-lists" >}}) are alerted. Because these metrics are applicable to multiple scenarios, they are configured project-wide, and then applied to test scenarios as needed. Multiple quality sensors can exist in a project, and multiple quality sensors can be assigned to any scenario.

### Defining a Quality Sensor

> To add a new quality sensor, go to the _Quality Sensors_ tab in _Configuration_ and click _New_ at the top right. You can define a name, description, and a number of failed executions for a number of consecutive executions (for example, 1/1 raises an alert on every failure, 3/3 raises an alert if the the scenario fails 3 times in a row, 3/10 raises an alert if 3 out of 10 consecutive executions failed).

Reviewer: ```suggestion
To add a new quality sensor, go to the _Quality Sensors_ tab in _Configuration_ and click _New_ at the top right. You can define a name, description, and a number of failed executions for a number of consecutive executions (for example, 1/1 raises an alert on every failure, 3/3 raises an alert if the scenario fails 3 times in a row, 3/10 raises an alert if 3 out of 10 consecutive executions failed).
```

{{< image src="xtc/monitoring_qualitySensors_details.png" >}}
Quality sensor detail view.
{{< /image >}}

In the quality sensor's details' _Configuration_ tab, you can add **metrics**, which are basically a set of conditions that must all be met or the scenario will fail. Select criteria from these groups:

* **General**: Check for execution errors or scenario failures.
* **Event**: Check the timing of [page load events]({{< relref "../../xlt/load-testing/manual/600-client-performance/#metrics-for-perceived-performance" >}}) (_DomContentLoaded_, _First Contentful Paint_, _First Paint_, _Load Event_).
* **Web Vitals**: Check [user experience criteria](https://web.dev/articles/vitals) (_Cumulative Layout Shift_, _First Contentful Paint_, _First Input Delay_, _Interaction to Next Paint_, _Largest Contentful Paint_, _Time to First Byte_).
* **XLT**: Check for basic runtime limits (for transactions, actions or requests) and request errors.

Depending on the selected metric, the threshold you will be asked to select will be just "on occurrence", a time in milliseconds or some more specific parameter. All defined metrics can be edited or deleted afterwards using their context menu. 

> Not meeting one of the defined thresholds may cause an alert to be raised (remember that this happens only after the number of consecutive failures you set before), which is then sent to the set of [**notification lists**]({{< relref "#notification-lists" >}}) defined below.  

Reviewer: A description how notification lists are assigned to a quality sensor is missing.
